server/leafnode.go

```diff
@@ -2360,7 +2360,7 @@ func (s *Server) leafNodeResumeConnectProcess(c *client) {
 			rURL := remote.getCurrentURL()
 
 			// Perform the client-side TLS handshake.
-			if resetTLSName, err := c.doTLSClientHandshake("leafnode", rURL, tlsConfig, tlsName, tlsTimeout); err != nil {
+			if resetTLSName, err := c.doTLSClientHandshake("leafnode", rURL, tlsConfig, tlsName, tlsTimeout, c.srv.getOpts().TLSPinnedCerts); err != nil {
```

Comment:
It is "client side", but that simply means that this is the soliciting side. I think that you should still provide the leafnodes pinned certs here, no?

Reply:
yes, fixed